modules/internal/ChapelLocale.chpl

```diff
@@ -182,34 +182,82 @@ module ChapelLocale {
   // Locale methods we want to have show up in chpldoc start here:
 
   /*
-    Get the hostname of this locale.
+    Get the hostname of the hardware associated with this locale.
+
+    The behavior of this method does not differ for calls made from
+    sub-locales (Unlike :proc:`locale.name`).
 
     :returns: the hostname of the compute node associated with the locale
     :rtype: string
+
   */
-  inline proc locale.hostname {
+  inline proc locale.hostname: string {
     return this._value.hostname;
   }
 
   /*
-    Get the name of this locale.  In practice, this is often the
-    same as the hostname, though in some cases (like when using
-    local launchers), it may be modified.
+    Get the name of this locale.
 
-    :returns: locale name
-    :rtype: string
+    In general, this method returns the same string as :proc:`locale.hostname`;
+    however, it can differ when called on a sub-locale (such as a GPU),
+    or when the program is executed in an oversubscribed manner.
+
+    The following table summarizes the various behaviors of this method (where
+    {hostname} is the string returned by `here.hostname`, and {id} is the locale's
+    unique integer ID):
+
+  .. list-table::
+    :widths: 25 22 22
+    :header-rows: 1
+
+    * - Configuration
+      - locale ( `here.name` )
+      - sub-locale ( `here.gpus[0].name` )
+    * - normal execution
+      - {hostname}
+      - {hostname}-GPU0
+    * - oversubscribed execution
+      - {hostname}
+      - {hostname}-GPU0
+    * - oversubscribed execution (gasnet*)
+      - {hostname}-{id}
+      - {hostname}-{id}-GPU0
+
+  .. note::
+
+    When launching in an oversubscribed manner with `CHPL_COMM=gasnet`
+    and one of the following configurations:
+
+    - `CHPL_COMM_SUBSTRATE=udp` & `GASNET_SPAWNFN=L`
+    - `CHPL_COMM_SUBSTRATE=smp`
+
+    the locale's integer ID will also be included in it's name.
+
+    More information about these environment variables can be found here: :ref:`readme-multilocale`
+
```

Comment:
The above note confused me more than helped. Your table already shows that the ID is included in the name, so this note made me think that there is more to it. Perhaps instead you could just have a pseudo-footnote like this:

(*) This behavior occurs when launching [... continue as above ...]